### PR TITLE
Get request with only params

### PR DESCRIPTION
**Pull request by hctym1995**
Some methods have been added to support the client to send GET requests with only params

### CODE REVIEW

**Comment by yhirose** [comment]
@hctym1995, thanks for the pull request. Could you add unit test cases for the additions?

**Comment by hctym1995** [comment]
Can you tell me where I should add it in test.cc?

**Comment by hctym1995** [comment]
Or some keywords

**Comment by yhirose** [comment]
@hctym1995, you could add a unit test like `TEST_F(ServerTest, GetMethodWithParams) { ... }`.

**Comment by hctym1995** [comment]
@yhirose  I have already added unit test function, please check if it is correct

File: test/test.cc [hunk 1]
```diff
@@ -2881,6 +2881,17 @@ TEST_F(ServerTest, GetStreamedChunkedWithGzip) {
   EXPECT_EQ(std::string("123456789"), res->body);
 }
 
+TEST_F(ServerTest, GetMethodWithParams) {
+  Params params;
+  params.emplace("test1", "param1");
+
+  auto res = cli_.Get("/getmethodwithparams", params);
```
Comment (yhirose) [inline test/test.cc:2888]: I don't see any corresponding `/getmethodwithparams` in `ServerTest::SetUp` where you have to test the given parameters.

You could refer to the following:
https://github.com/yhirose/cpp-httplib/blob/301faa074c4a0fa1dbe470dfb4f77912caa1c57f/test/test.cc#L1448-L1456

File: test/test.cc [hunk 2]
```diff
@@ -2881,6 +2881,17 @@ TEST_F(ServerTest, GetStreamedChunkedWithGzip) {
...
+  ASSERT_TRUE(res);
+  EXPECT_EQ(200, res->status);
+  EXPECT_EQ(std::string("123456789"), res->body);
```
Comment (yhirose) [inline test/test.cc:2891]: This doesn't actually test the parameters at all...

**Comment by yhirose** [comment]
@hctym1995, the unit test you added doesn't really test the given parameters. Could you follow my comments in the review?

**Review by Supsource — APPROVED** [review]

**Comment by yhirose** [comment]
@hctym1995, I just compiled your changes with `test/test.cc`, and got a number of compile errors. It seems like the changes are not really tested... I appreciated your efforts though, I rejected the change for this reason. Thank you for your understanding.

image

**Pull request closed without merging**